com/metacube/SCF/Assignment1/ShoppingCart.java

```diff
@@ -0,0 +1,107 @@
+package GET2018.com.metacube.SCF.Assignment1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Copyright (c) 2018 Metacube.com. All rights reserved.
+ * This class is created for declaring various shopping cart function.
+ * @author Chirag Jain
+ */
+public class ShoppingCart {
+
+	List<Product> products = new ArrayList<Product>();
+    Promotion promotion;
+        
+    /**
+     * this method adds an item to the cart. If an item already exists, its quantity is updated
+     * @param id 
+     * @param price
+     * @param quantity
+     */
+    public void addItem(int id, double price, int quantity)
+    {
+        boolean update = false;
+        for(Product item : products)
+        {
+            if(item.getId()==id)
+            {
+                update = true;
+                int q = item.getQuantity();
+                item.setQuantity(quantity + q);
+                break;
+            }
+        }
+        if(!update)
+        {
+        	Product item = new Product(id, price, quantity);
+            products.add(item);
+        }
+    }
+        
+   /**
+    * this method removes an item from the cart
+    * @param id is the product id of the product to be removed
+    */
+    public void removeItem(int id)
+    {
+        for(Product item:products)
+        {
+            if(id==item.getId())
+            {
+                products.remove(item);
+                break;
+            }
+        }
+    }
+      
+    /**
+     * this method generates the bill of the cart items and also applies the discount if applicable
+     * @param promoCode is the promo code as provided by user
+     */
+    public void generateBill(String promoCode)
+    {
+        double sum=0;
+        System.out.println("Product id\t\tprice\t\tquantity\ttotal");
+        for(Product item:products)
+        {
+            item.showProducts();
+            sum += item.total();
+        }
+            
+       promotion = new FixedOrderPromotion();
+       promotion.setMinimumPrice(200);
```

Comment:
You can have a list of products in FixedProductPromotion class and mimimum order constant in FixedOrderPromotion class. You can iterate the list of products in shopping cart class and compare it with the product list in FixedProductPromotion class and thus segregate the products. Then apply the suitable discount.